kms/mackms/mackms.go

```diff
@@ -48,6 +50,11 @@ const Scheme = string(apiv1.MacKMS)
 // the keys.
 var DefaultTag = "com.smallstep.crypto"
 
+// UseDataProtectionKeychain defines if mackms will use the Data Protection
+// keychain by default. This option requires the application to be code-signed
+// with the appropriate entitlements.
+var UseDataProtectionKeychain = false
```

Comment:
Instead of setting this on the package level, can it be provided when instantiating an instance instead? That's closer to how we do things for e.g. the TPM KMS with CAPI feedthrough.

Reply:
Additional, what benefit is there to allow specifying the default behavior? Shouldn't we always prefer to use `UseDataProtectionKeychain` based on where the key resides? There should never be any reason to store the certificate in a file-based keychain when the key is on the data-protection keychain, and vice-versa.

Reply:
@hslatman: Fixed with caf7865. 

> Shouldn't we always prefer to derive UseDataProtectionKeychain based on where the key resides?

@joshdrake: We are doing that, if you don't pass any keychain to `StoreCertificate` or `StoreCertificateChain` it will look for the key and store it in the proper location.